### PR TITLE
Clean up existing header guards, add missing ones

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
* Add a `BLOSC_` prefix to all headers to avoid collisions.
* Standardise header guards based on the path of the header file.

### CODE REVIEW

File: plugins/filters/bytedelta/bytedelta.h [hunk 1]
```diff
@@ -8,7 +8,9 @@
   See LICENSE.txt for details about copyright and rights to use.
 **********************************************************************/
 
-#pragma once
+#ifndef BLOSC_PLUGINS_FILTERS_BYTEDELTA_BYTEDELTA_H
+#define BLOSC_PLUGINS_FILTERS_BYTEDELTA_BYTEDELTA_H
```
Comment (FrancescAlted) [inline plugins/filters/bytedelta/bytedelta.h:12]: Duplicated BYTEDELTA substring?

File: plugins/filters/ndcell/ndcell.h [hunk 2]
```diff
@@ -8,8 +8,8 @@
   See LICENSE.txt for details about copyright and rights to use.
 **********************************************************************/
 
-#ifndef B2ND_NDCELL_H
-#define B2ND_NDCELL_H
+#ifndef BLOSC_PLUGINS_FILTERS_NDCELL_NDCELL_H
+#define BLOSC_PLUGINS_FILTERS_NDCELL_NDCELL_H
```
Comment (FrancescAlted) [inline plugins/filters/ndcell/ndcell.h:12]: Ditto

File: plugins/filters/ndmean/ndmean.h [hunk 3]
```diff
@@ -8,8 +8,8 @@
   See LICENSE.txt for details about copyright and rights to use.
 **********************************************************************/
 
-#ifndef B2ND_NDMEAN_H
-#define B2ND_NDMEAN_H
+#ifndef BLOSC_PLUGINS_FILTERS_NDMEAN_NDMEAN_H
+#define BLOSC_PLUGINS_FILTERS_NDMEAN_NDMEAN_H
```
Comment (FrancescAlted) [inline plugins/filters/ndmean/ndmean.h:12]: Ditto

**Review by FrancescAlted — COMMENTED** [review]
Just a minor thing in the header guards naming

File: plugins/filters/ndmean/ndmean.h (hunk 3 above)
Comment (DimitriPapadopoulos) [reply plugins/filters/ndmean/ndmean.h:12]: Same

File: plugins/filters/bytedelta/bytedelta.h (hunk 1 above)
Comment (DimitriPapadopoulos) [reply plugins/filters/bytedelta/bytedelta.h:12]: Duplicate as the path itself:
`plugins/filters/`**`bytedelta/bytedelta.h`**

File: plugins/filters/ndcell/ndcell.h (hunk 2 above)
Comment (DimitriPapadopoulos) [reply plugins/filters/ndcell/ndcell.h:12]: Same

File: plugins/filters/bytedelta/bytedelta.h (hunk 1 above)
Comment (FrancescAlted) [reply plugins/filters/bytedelta/bytedelta.h:12]: I see the pattern now.  It is ok with me.

**Review by FrancescAlted — APPROVED** [review]
Looks good to me

**Comment by FrancescAlted** [comment]
Thanks @DimitriPapadopoulos !

**Pull request merged**